src/raftstore/store/fsm/apply.rs

```diff
@@ -2547,27 +2531,25 @@ impl ApplyFsm {
         }
 
         // if it is already up to date, no need to catch up anymore
-        if catch_up_logs.logs_up_to_date.load(Ordering::SeqCst) == 0 {
```

Comment:
Why skip the check now?

Reply:
note that we don't set `logs_up_to_date` in `on_ready_prepare_merge` anymore, it is updated only after `apply_index >= catch_up_logs.merge.get_commit()`. so here `logs_up_to_date` is always zero